webpack.common.js

```diff
@@ -0,0 +1,62 @@
+const path = require('path');
+const HTMLWebpackPlugin = require('html-webpack-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+
+module.exports = {
+    context: path.resolve(__dirname, 'src'),
+    entry: {
+        app: './index.jsx'
+    },
+    output: {
+        path: path.resolve(__dirname, 'dist'),
+        filename: '[name].[fullhash].js',
+        publicPath: '/'
+    },
+    resolve: {
+        extensions: ['.js', '.jsx']
+    },
+    plugins: [
+        new CleanWebpackPlugin(),
+        new HTMLWebpackPlugin({
+            template: './index.html'
+        }),
+        new MiniCssExtractPlugin({
+            filename: '[name].[fullhash].css'
```

Comment:
am I right that thiss will minify css for dev mode as well as for prod? Do we really need it?

Reply:
@lotrien, this plugin extracts CSS into separate files.

Reply:
oh, thanks for clarification. somewhat confusing name